### ISSUE TITLE
Deleting documents not obeying the 'ignore_not_found' parameter

### ISSUE THREAD

**Issue by harro33** [typesense/typesense-js#217]
## Description

When deleting documents via the API, passing the 'ignore_not_found' parameter still results in an error response

## Steps to reproduce

1. Manually delete a document on Typesense

2. Try to delete the same document via the API using code something like this:

```
const response = await this.client
  .collections(collectionName)
  .documents(documentId)
  .delete({'ignore_not_found': true});
return response;
```

## Expected Behavior

A response, but not an error.

## Actual Behavior

Error: "Could not find a document with id: ..."

## Metadata

**Typesense Version**: 26.0 and 27.0.rc25
**Typesense-js Version**: 1.8.2

**Comment by tharropoulos** [comment]
Could you post a reproducible example? This seems a bit vague

**Comment by harro33** [comment]
> Could you post a reproducible example? This seems a bit vague

Hi @tharropoulos. Sorry if this wasn't enough information, but I'm not sure what else I should have provided. @kishorenc has my contact details and I am happy to get on a video call to show you what is happening if that would help? It could be something I am doing wrong, or something specific to my setup, so a video call might help get to the bottom of this faster.

**Comment by kishorenc** [comment]
@harro33 Sorry about that, I will sync up with @tharropoulos to add additional context on the debugging steps we tried :)

**Comment by harro33** [comment]
> @harro33 Sorry about that

All good. As I say, happy to jump on another video call to help.

**Comment by tharropoulos** [comment]
Just submitted a PR, the delete method on documents didn't include a parameter for the query parameters passed to the actual API call, so that's why it resulted in the errors you mentioned.  You can test it yourself and see if it resolves your issues! It now works on a manual test I tried as well

**Comment by tharropoulos** [comment]
@harro33 If the change seems to resolve your issue, please mark this issue as solved. If it didn't, we're more than willing to look more into it!

**Comment by kishorenc** [comment]
@tharropoulos 

I will work with Jason to do a point release of the library so that it's easy to test.

**Comment by jasonbosco** [comment]
Published this in v2.0.0-0
